File: 08.BrainTwist脑筋急转弯/leetcode_1752_check.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution {
public:
    bool check(vector<int>& nums) {
        int cnt = 0;
        for (int i=0; i<nums.size()-1; i++){
            if (nums[i] > nums[i+1]){
                cnt++;
            }
        }
        return cnt == 0 || (cnt == 1 && nums.back() <= nums[0]);
    }
};

int main(){
    Solution s;
    vector<int> nums = {3,4,5,1,2};
    cout << s.check(nums) << endl;
}

File: 08.BrainTwist脑筋急转弯/leetcode_1752_check.py
from typing import List

class Solution:
    def check(self, nums: List[int]) -> bool:
        cnt = 0
        for i in range(len(nums)-1):
            if nums[i] > nums[i+1]:
                cnt += 1
        return cnt == 0 or (cnt == 1 and nums[-1] <= nums[0])

if __name__ == "__main__":
    s = Solution()
    nums = [3,4,5,1,2]
    print(s.check(nums))
